shared/src/main/scala/fr/hmil/scalahttp/Protocol.scala

```diff
@@ -19,5 +19,9 @@ object Protocol {
   val HTTP = Protocol("HTTP")
   val HTTPS = Protocol("HTTPS")
 
-  implicit def fromString(name: String): Protocol = new Protocol(name.toUpperCase)
+  implicit def fromString(name: String): Protocol = name.toUpperCase match {
```

Comment:
I'm using this in the URL parsing code but I guess I could remove the implicit modifier as it doesn't make sense for the user to rely on this function.